src/transformers/generation/logits_process.py

```diff
@@ -2219,3 +2219,143 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
             scores_processed = torch.where(do_early_stop, early_stop_scores, scores)
 
         return scores_processed
+
+
+class WatermarkLogitsProcessor(LogitsProcessor):
+    r"""
+    Logits processor for watermarking generated text. The processor modifies model output scores by adding a small bias to
+    randomized set of "green" tokens before generating the next token. "Green" tokens selection process depends on the
+    `seeding_scheme` used. The code was based on the [original repo](https://github.com/jwkirchenbauer/lm-watermarking/tree/main).
```

Comment:
what are the red tokens used for then?

Reply:
They are not used, but we mention it just to make a contrast between "green" (allowed and boost by biasing) and "red" (not allowed so no bias added)

Reply:
i removed the mention of 'red' all-together, so as to not confuse anyone :)